From 4d86a8f3f690ee9bb9429e17cc03856c6c2dc760 Mon Sep 17 00:00:00 2001
From: Shaun Ruffell <sruffell@digium.com>
Date: Mon, 22 Dec 2014 11:21:28 -0600
Subject: [PATCH] dahdi: struct file.f_dentry macro was removed in kernel 3.19

This is necessary to build against kernel version 3.19 since commit
(78d28e651f97866d608d9b41 "kill f_dentry macro") [1]

[1] https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/commit/?id=78d28e651

Signed-off-by: Shaun Ruffell <sruffell@digium.com>
Signed-off-by: Russ Meyerriecks <rmeyerriecks@digium.com>
---
 drivers/dahdi/dahdi-base.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/dahdi/dahdi-base.c b/drivers/dahdi/dahdi-base.c
index 0892734..f2caad0 100644
--- a/drivers/dahdi/dahdi-base.c
+++ b/drivers/dahdi/dahdi-base.c
@@ -98,7 +98,11 @@
 #define chan_to_netdev(h) ((h)->hdlcnetdev->netdev)
 
 /* macro-oni for determining a unit (channel) number */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 20)
 #define	UNIT(file) MINOR(file->f_dentry->d_inode->i_rdev)
+#else
+#define	UNIT(file) MINOR(file->f_path.dentry->d_inode->i_rdev)
+#endif
 
 EXPORT_SYMBOL(dahdi_transcode_fops);
 EXPORT_SYMBOL(dahdi_init_tone_state);
-- 
2.3.4

